src/components/publishing/imageset_preview_classic.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from "react"
+import styled from "styled-components"
+
+import IconImageset from "./icons/icon_imageset"
+
+import Fonts from "./fonts"
+
+const Wrapper = styled.div`
+  max-width: 580px;
+  width: 100%;
+  height: 150px;
+  display: flex;
+`
+const Text = styled.div`
+  ${Fonts.avantgarde("s11")}
+  line-height: 1.35em;
+  margin: 0;
+`
+const Remaining = styled.div`
+  min-width: 50px;
+  padding: 10px;
+  flex: 1;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  border: 1px solid #e5e5e5;
+  text-align: center;
+`
+const IconContainer = styled.div`
+  width: 32px;
+  margin-bottom: 10px;
+  position: relative;
+`
+
+class ImagesetPreviewClassic extends Component<any, any> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      visibleImages: this.getVisibleImages(this.props.images),
+    }
+  }
+
+  getVisibleImages(images) {
+    let widths = []
+    let hidden = 0
+    images.map((item, i) => {
+      const adjustedWidth = 150 * item.width / item.height
+      widths.push(adjustedWidth)
+      const total = widths.reduce((a, b) => a + b, 0)
+      const margins = widths.length * 10
+      if (total + margins + 50 > 560) {
+        hidden = hidden + 1
```

Comment:
Would be nice to move all these magic numbers to the top of the file as self-describing constants.